data_steward/cdr_cleaner/clean_cdr.py

```diff
@@ -46,9 +47,71 @@
 import cdr_cleaner.manual_cleaning_rules.remove_operational_pii_fields as operational_pii_fields
 import cdr_cleaner.manual_cleaning_rules.update_questiona_answers_not_mapped_to_omop as map_questions_answers_to_omop
 from constants.cdr_cleaner.clean_cdr import DataStage as stage
+from constants.cdr_cleaner import clean_cdr as cdr_consts
 
 LOGGER = logging.getLogger(__name__)
 
+PROJECT_ID = 'project_id'
+DATASET_ID = 'dataset_id'
+SANDBOX_DATASET_ID = 'sandbox_dataset_id'
+
+
+class CleanCdrIllegalArgumentError(ValueError):
+    pass
+
+
+def add_module_info_decorator(query_function,
+                              project_id,
```

Comment:
As defined, this will break.  I tested it with the patch I am attaching.  It raises your `CleanCdrIllegalArgumentError`.

[improved_logging_breaks.txt](https://github.com/all-of-us/curation/files/4182624/improved_logging_breaks.txt)

Essentially, the above patch assumes every query module is using a keyword based approach for parameters.  They are not.  This approach requires all the query modules name their parameters the same way.  Running this patch with `PYTHONPATH=.:$PYTHONPATH python cdr_cleaner/clean_cdr.py -d 'deid_base'`, you can see the module is expecting `sandbox_dataset` but is receiving `sandbox_dataset_id`.  This fails and is not able to seamlessly integrate with each existing query module.

[improved_logging_works.txt](https://github.com/all-of-us/curation/files/4182625/improved_logging_works.txt)
The above patch makes use of positional and keyword parameter arguments.  It uses the `*` and `**` prefixes.  It lets the query modules use whatever naming convention they prefer. It also allows you to remove some unnecessary global variables, dictionaries, and checks.  You can test it with `PYTHONPATH=.:$PYTHONPATH python cdr_cleaner/clean_cdr.py -d 'deid_base'` also.

Of course, fixing this will require updating your tests.  Let me know what you think.

FYI:  to apply a patch from file from the command line:
```
cd <wherever your project is>/curation/data_steward
git apply <path to downloaded patch file>
```
then you can activate your python environment and run the python command on your patched file to see the break and the working file.

Reply:
Thanks for providing those patches and instructions on how to apply those! This is really cool and it is much much easier to read and understand once I get around the positional the keyword arguments. I was trying to solve this issue by introducing the check but this was totally unnecessary.